src/decays/experimental_constraints.cpp

```diff
@@ -0,0 +1,579 @@
+#include <gsl/gsl_errno.h>
+#include <boost/math/distributions/chi_squared.hpp>
+
+#include <algorithm>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <functional>
+#include <random>
+
+#ifdef ENABLE_HIGGSTOOLS
+#include "Higgs/Predictions.hpp"
+#include "Higgs/Bounds.hpp"
+#include "Higgs/Signals.hpp"
+namespace HP = Higgs::predictions;
+#endif
+
+#ifdef ENABLE_LILITH
+#include <Python.h>
+#include "lilith.h"
+#include "lilith.c"
+#endif
+
+namespace flexiblesusy {
+
+namespace {
+
+// Whether to calculate the X->gaga and X->gg decay widths in terms of the
+// effective tree-level couplings (calcHgamgam=calcggH=true) or by rescaling
+// the X->gaga and X->gg decay by the squared of the effective gamgam and gg
+// couplings (calcHgamgam=calcggH=false). True would make sense only if we
+// would not compute those loop-induced decays ourself.
+constexpr bool calcggH     = false;
+constexpr bool calcHgamgam = false;
+
+// relative BSM Higgs-like state mass uncertainty
+// example: 0.03 means 3% uncertainty
+constexpr double relMassError = 0.03;
+
+#ifdef ENABLE_HIGGSTOOLS
+// Ref. model for computing brs and xsections on the HiggsTools side
+constexpr auto refModel = HP::ReferenceModel::SMHiggsInterp;
+
+double minChi2SM_hs(const double mhSM, std::string const& higgssignals_dataset) {
+   const auto signals = Higgs::Signals {higgssignals_dataset};
+
+   auto pred = Higgs::Predictions();
+   auto& s = pred.addParticle(HP::BsmParticle("hSM", HP::ECharge::neutral, HP::CP::even));
+   auto effc = HP::scaledSMlikeEffCouplings(1.0);
+   s.setMass(mhSM);
+   s.setMassUnc(0.);
+   effectiveCouplingInput(
+      s, effc,
+      refModel,
+      calcggH, calcHgamgam
+   );
+   return signals(pred);
+}
+
+void print_effc(double mass, HP::NeutralEffectiveCouplings const& effC) {
+   std::cout << "Effective couplings for particle of mass " << mass << '\n';
+   std::cout << "dd     " << effC.dd << std::endl;
+   std::cout << "uu     " << effC.uu << std::endl;
+   std::cout << "ss     " << effC.ss << std::endl;
+   std::cout << "cc     " << effC.cc << std::endl;
+   std::cout << "bb     " << effC.bb << std::endl;
+   std::cout << "tt     " << effC.tt << std::endl;
+   std::cout << "ee     " << effC.ee << std::endl;
+   std::cout << "mumu   " << effC.mumu << std::endl;
+   std::cout << "tautau " << effC.tautau << std::endl;
+
+   std::cout << "WW     " << effC.WW << std::endl;
+   std::cout << "ZZ     " << effC.ZZ << std::endl;
+   std::cout << "gamgam " << effC.gamgam << std::endl;
+   std::cout << "Zgam   " << effC.Zgam << std::endl;
+   std::cout << "gg     " << effC.gg << std::endl;
+   std::cout << "lam     " << effC.lam << std::endl;
+}
+#endif
+
+#ifdef ENABLE_LILITH
+double minChi2SM_lilith(const double mhSM) {
+
+   Py_Initialize();
+   char experimental_input[] = "";
+   // Creating an object of the class Lilith: lilithcalc
+   PyObject* lilithcalc2 = initialize_lilith(experimental_input);
+
+   char XMLinputstring[6000] = "";
```

Comment:
Sorry to be pedantic, but I'm afraid of filling these fixed-size buffers. Please use a `std::string` or `std::sstream` instead, which is dynamically re-sized. We can always use `.c_str()` to get a C-style character array out, if we need it at the end.

Reply:
Done in 4f9b9025de97dd9aaac54036b32101e380fe7fc7.

Reply:
Excellent, thanks!